Library/Homebrew/download_strategy.rb

```diff
@@ -419,10 +419,12 @@ def fetch(timeout: nil)
         # Authorization is no longer valid after redirects
         meta[:headers]&.delete_if { |header| header.start_with?("Authorization") } if is_redirection
 
-        # The cached location is no longer fresh if Last-Modified is after the file's timestamp
-        if cached_location.exist? && last_modified && last_modified > cached_location.mtime
-          use_cached_location = false
-        end
+        # The cached location is no longer fresh if either:
+        # - Last-Modified value is newer than the file's timestamp
+        # - Content-Length value is different than the file's size
+        newer_last_modified = last_modified && last_modified > cached_location.mtime
+        different_file_size = file_size && file_size != cached_location.size
```

Comment:
Do you need to check `file_size` here, or is this equivalent?
```suggestion
        different_file_size = file_size != cached_location.size
```

Reply:
The absence of `file_size` would imply that a `Content-Length` header was not provided by the server.

Reply:
right, but the logic of `file_size != cached_location.size` should still be equivalent, because `cached_location.size` is not nilable, if I followed along correctly

Reply:
Sorry, it's early and i'm an idiot